Helios/Interfaces/Profile/ProfileInterface.cs

```diff
@@ -133,13 +154,122 @@ protected override void DetachFromProfileOnMainThread(HeliosProfile oldProfile)
+                int blank = e.Value.StringValue.IndexOf(" ");
+                if (matches.Count == 0)
+                {
+                    //  There is nothing enclosed in double-quotes, so we assume the executable is before the first space and any arguments follow the first space
+
+                    if (blank > 0)
+                    {
+                        expandedPath = Environment.ExpandEnvironmentVariables(e.Value.StringValue.Substring(0, blank));
+                        expandedArgs = Environment.ExpandEnvironmentVariables(e.Value.StringValue.Substring(blank + 1));
+                    }
+                    else
+                    {
+                        expandedPath = Environment.ExpandEnvironmentVariables(e.Value.StringValue);
+                        expandedArgs = "";
+                    }
+                }
+                else
+                {
+                    int matchCursor = 0;
+                    foreach (Match matchItem in matches)
+                    {
+                        if (matchItem.Equals(matches[0]))
+                        {
+                            if (matchItem.Index == 0)
+                            {
+                                expandedPath = Environment.ExpandEnvironmentVariables(matchItem.Groups[1].ToString());
+                                matchCursor = matchItem.Length;
+                            }
+                            else
+                            {
+                                expandedPath = Environment.ExpandEnvironmentVariables(e.Value.StringValue.Substring(0, blank));
+                                expandedArgs = Environment.ExpandEnvironmentVariables(e.Value.StringValue.Substring(blank, matchItem.Index - blank));
+                                expandedArgs += " " + Environment.ExpandEnvironmentVariables(matchItem.ToString());
+                                matchCursor = matchItem.Index + matchItem.Length;
+                            }
+                        }
+                        else
+                        {
+                            if (matchItem.Index == matchCursor)
+                            {
+                                expandedArgs += " " + Environment.ExpandEnvironmentVariables(matchItem.ToString());
+                            }
+                            else
+                            {
+                                expandedArgs += Environment.ExpandEnvironmentVariables(e.Value.StringValue.Substring(matchCursor, matchItem.Index - matchCursor));
+                                matchCursor = matchItem.Index + matchItem.Length;
+                                expandedArgs += " " + Environment.ExpandEnvironmentVariables(matchItem.ToString());
+                            }
+                        }
+                        Logger.Debug("Double Quoted item found {ItemText}", matchItem.ToString());
```

Comment:
just "matchItem"   as ToString() is called by the logger

Reply:
removed